app/controllers/questions/questions_controller.rb

```diff
@@ -99,6 +100,26 @@ def redirect_if_completed_intake_present
       end
     end
 
+    def redirect_if_no_intake
+      if current_intake.present?
+        # Assign the global id of the state_file_intake of the session to be that of the current_intake
+        # This attempts to prevent the weird timeout issues we have been experiencing in the flow.
+        # Where the session's intake does match the true current one and ends up timing out erroneously.
+        session[:state_file_intake] = "gid://vita-min/#{current_intake.class}/#{current_intake.id}"
+        # Sign out from previous intake if they differ from current_intake
+        # TODO
```

Comment:
I don't believe this would change anything and I'm concerned about including it. `current_intake` essentially checks if the value in the session matches the value on line 108. Line 108 sets the session value to the same value that has just been confirmed by line 105.

Reply:
This logic makes sure that the session intake matches the current_intake, because as you suggested the intake on the session may be a different one that is bound to time out or already has.

Reply:
This code has been deleted